### DISCUSSION TITLE
p/cloudflared-tunnel-in-openwrt/

### DISCUSSION THREAD

**Discussion by giscus** [donhector/donhector.github.io#22] [Giscus]
# p/cloudflared-tunnel-in-openwrt/

Running the cloudflared binary as a service in OpenWRT

https://hmolina.dev/p/cloudflared-tunnel-in-openwrt/

**Comment by sfdzkj** [comment]
非常感谢您的分享。你的文章对我帮助很大

**Reply by donhector** [reply]
Glad it helped!

**Comment by milindpatel63** [comment]
i am running this on a mi wifi mini router running latest openwrt

I am getting this error

```
Sat Nov 12 03:21:39 2022 daemon.err cloudflared[4798]: 2022-11-11T21:51:39Z WRN The user running cloudflared process has a GID (group ID) that is not within ping_group_range. You might need to add that user to a group within that range, or instead update the range to encompass a group the user is already in by modifying /proc/sys/net/ipv4/ping_group_range. Otherwise cloudflared will not be able to ping this network error="Group ID 0 is not between ping group 1 to 0"
Sat Nov 12 03:21:39 2022 daemon.err cloudflared[4798]: 2022-11-11T21:51:39Z WRN ICMP proxy feature is disabled error="cannot create ICMPv4 proxy: Group ID 0 is not between ping group 1 to 0 nor ICMPv6 proxy: socket: permission denied"
```
Any solutions?

**Reply by donhector** [reply]
My guess is that your are running `cloudflared` using an unprivileged user. If that's the case, [this](https://opennms.discourse.group/t/how-to-allow-unprivileged-users-to-use-icmp-ping/1573) post might lead you in the right direction to include your user's group in the list of allowed groups.

**Reply by yonaesp** [reply]
I lost many hours until I found the solution, but here I share it:
1. Create a file with this name: 99-fix-buffer-and-ping.conf in the folder: /etc/sysctl.d/
2. Add this inside and save the file.:
<sup>net.core.rmem_max=2500000
net.ipv4.ping_group_range=0 2147483647</sup>
3. Execute this command from SSH:<sup>sysctl -p /etc/sysctl.d/99-fix-buffer-and-ping.conf</sup>

Also remember, it is now possible to install from opkg, remember to follow the steps on github to move the corresponding configuration files once installed.

**Reply by donhector** [reply]
Thanks for sharing 👌

**Comment by thematrixdev** [comment]
Thanks for your sharing. It works on my OpenWRT v21 router.

By the way, I have two WAN (ISP) access. I want the both IP addresses are included for the tunnel. Is it possible?

Thanks.

**Comment by spereg53** [comment]
i try to install on modem gli-net GL-E750C4 that have openwrt 19.07.8 and can not connect ,
give me this :Thu Apr 27 04:14:02 2023 daemon.info procd: Instance cloudflared::instance1 s in a crash loop 7 crashes, 0 seconds since last crash
can you helpme,thanks in advanced

**Reply by donhector** [reply]
You should look at your OpenWrt logs for any clues. Use the `logread` command for that.
